acceptance/acceptance_test.go

```diff
@@ -262,7 +262,7 @@ func runTest(t *testing.T, dir, coverDir string, repls testdiff.ReplacementsCont
 	// 2. The test is configured to record requests and assert on them. We need
 	//    a duplicate of the default server to record requests because the default
 	//    server otherwise is a shared resource.
-	if len(config.Server) > 0 || config.RecordRequests {
+	if cloudEnv == "" && (len(config.Server) > 0 || config.RecordRequests) {
```

Comment:
We use this conditions `cloudEnv == ""` in many places already to detect that it runs in Cloud, can we extract it to separate function? Makes it easier to read and refactor in case logic ever changes?

Reply:
+1

The switch between local-only mode and real mode is binary and strict. Makes sense to centralize the logic that makes this decision.

Reply:
To me CLOUD_ENV / cloudEnv is part of the API of the test runner / environment. It is centralized already.

We can add another alias, e.g. IsRunningOnCloud variable or IsRunningOnCloud() function, but to me that adds nothing but extra level of indirection and extra name to remember.

It is possible that we can do a better job here to group all cloud vs local logic in one place, I'll take a look.

Reply:
This switch might become even more complex when we will need to differentiate between UC enabled and non-UC ones, so this would be another variable to consider